WooCommerce/WooCommerceTests/POS/Controllers/PointOfSaleOrderControllerTests.swift

```diff
@@ -128,25 +145,35 @@ struct PointOfSaleOrderControllerTests {
         ])
     }
 
+    @available(iOS 17.0, *)
     @Test func syncOrder_with_order_sync_failure_sets_orderState_syncing_then_error() async throws {
         // Given
+        let sut = PointOfSaleOrderController(orderService: mockOrderService,
+                                             receiptService: mockReceiptService)
         mockOrderService.orderToReturn = nil
 
-        var cancellables = Set<AnyCancellable>()
-        var orderStates: [PointOfSaleInternalOrderState] = []
-        await confirmation() { confirmation in
-            // We can use `withObservationTracking` when we move to @Observable
-            sut.orderStatePublisher.collect(3)
-                .sink { orderState in
-                    orderStates.append(contentsOf: orderState)
+        var orderStates: [PointOfSaleInternalOrderState] = [sut.orderState]
+        var orderStateAppendTask: Task<Void, Never>? = nil
+        await confirmation(expectedCount: 2) { confirmation in
+            @Sendable func observeOrderState() {
+                withObservationTracking {
+                    _ = sut.orderState
+                } onChange: {
+                    orderStateAppendTask = Task { @MainActor in
+                        orderStates.append(sut.orderState)
+                    }
                     confirmation()
+                    observeOrderState()
                 }
-                .store(in: &cancellables)
+            }
+            observeOrderState()
 
             // When
             await sut.syncOrder(for: [makeItem()], retryHandler: {})
         }
 
+        await orderStateAppendTask?.value
```

Comment:
A clever workaround 👍 

I think there's value in these tests, not just testing different states in isolation but confirming the expected sequence of the flow. Of course, we shouldn't make the whole test suite like this.

I'm sure us or the community will come up with decent helper functions for this behavior, just to make it easier to write and read tests. Not much information still both on Swift Testing and `@Observation` and almost nothing on using them both.

Reply:
That's nice @staskus seeing this way I find it easier to understand, specially the bit around _"triggering the action after the confirmations have been observed"_, so we assert all the expected states.